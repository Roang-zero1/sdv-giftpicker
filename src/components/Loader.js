import React, { Component } from 'react';
import { Container } from 'reactstrap';
import './Loader.css';

class Loader extends Component {
  render() {
    return (
      <Container className={'loader-wrapper'}>
        <h2>Loading</h2>
        <p>Loading Data please wait</p>
        <div className={'loader'} />
      </Container>
    );
  }
}

export default Loader;
