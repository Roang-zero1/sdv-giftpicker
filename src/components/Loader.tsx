import * as React from 'react';
import { Component } from 'react';

import { Container } from 'reactstrap';
import styled from 'styled-components';

const Wrapper = styled(Container)`
  height: 14em;
`;

const LoaderDiv = styled.div`
  direction: ltr;
  text-indent: -9999em;
  margin: 3em auto;
  position: relative;
  font-size: 11px;
  transform: translateZ(0);

  &,
  &::after,
  &::before {
    background: #222222;
    animation: load1 1s infinite ease-in-out;
    animation-delay: -160ms;
    width: 1em;
    height: 4em;
  }

  &::before,
  &::after {
    position: absolute;
    top: 0;
    content: '';
  }

  &::before {
    left: -1.5em;
    animation-delay: -320ms;
  }

  &::after {
    left: 1.5em;
    animation-delay: 0ms;
  }
  @keyframes load1 {
    0%,
    80%,
    100% {
      box-shadow: 0 0;
      height: 4em;
    }
    40% {
      box-shadow: 0 -2em;
      height: 5em;
    }
  }
`;

class Loader extends Component {
  public render() {
    return (
      <Wrapper>
        <h2>Loading</h2>
        <p>Loading Data please wait</p>
        <LoaderDiv />
      </Wrapper>
    );
  }
}

export default Loader;
