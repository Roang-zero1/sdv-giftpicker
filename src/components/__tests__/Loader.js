import Loader from '../Loader';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('components/Loader --- Shallow render component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loader />);
  });

  it('should render the one component', () => {
    expect(wrapper.length).toEqual(1);
  });
});

describe('components/Loader --- Snapshot', () => {
  it('should be the same as the last snapshot', () => {
    const renderValue = renderer.create(<Loader />).toJSON();

    expect(renderValue).toMatchSnapshot();
  });
});
