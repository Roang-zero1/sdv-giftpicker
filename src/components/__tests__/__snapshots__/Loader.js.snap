// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Loader --- Snapshot should be the same as the last snapshot 1`] = `
<div
  className="loader-wrapper container"
>
  <h2>
    Loading
  </h2>
  <p>
    Loading Data please wait
  </p>
  <div
    className="loader"
  />
</div>
`;
